### ISSUE TITLE
Adaptive loss: update every n epochs

### ISSUE THREAD

**Issue by rick433** [Gao-Group/atomdnn#1]
Hello, first of all, I'd like to thank you for your work. Inspired by your [interesting preprint](https://arxiv.org/pdf/2403.18122), I am currently implementing Algorithm 1 in order to test the adaptive loss weight strategy for my research. Do you have experience on what typical values for the parameter `n` lead to decent results? Figure 3 suggests that you set `n=100`?

**Comment by docampomill** [comment]
Howdy! Thank you for your interest in our work. You're right, we used `n=100` for the model in the paper. This parameter enables averaging `n` entries before recomputing $\alpha_k^i$ values, since in many cases the loss function of a neural network fluctuates. 

I have a personal rule of thumb for selection of such parameter. Plotting trainning and validation curves using moving averages (i.e. [Moving Average](https://en.wikipedia.org/wiki/Moving_average)) and finding a window size `k` that reduces the noise, while still capturing the trends. That same value should enable optimal results for `n`. Note that if the loss functions don't fluctuate much, `n=1` should be a good selection.

**Comment by rick433** [comment]
Thanks a lot for your quick response! Looking at the moving average seems like a good heuristic for choosing `n`. I hadn't thought about that. Cheers!
